reana_server/fetcher.py

```diff
@@ -82,6 +95,63 @@ def _clean_workflow_name(name: str) -> str:
         """
         return REGEX_CHARS_TO_REPLACE.sub("-", name).strip("-")
 
+    @staticmethod
+    def _download_file(url: str, output_path: str):
+        """Download the given URL.
+
+        This method also checks that the file to be downloaded does not exceed the
+        maximum file size allowed (``FETCHER_MAXIMUM_FILE_SIZE``).
+
+        :param url: URL of the file to be downloaded.
+        :param output_path: Path where the file will be downloaded to.
+        """
+
+        def write_to_file(response: requests.Response, output_path: str) -> int:
+            """Write the response content to the given file.
+
+            :param response: Response to be written to the output file.
+            :param output_path: Path to the output file.
+            :returns: Number of bytes read from the response content.
+            """
+            read_bytes = 0
+            with open(output_path, "wb") as output_file:
+                # Use the same chunk size of `urlretrieve`
+                for chunk in response.iter_content(chunk_size=1024 * 8):
+                    read_bytes += len(chunk)
+                    output_file.write(chunk)
+                    if read_bytes > FETCHER_MAXIMUM_FILE_SIZE:
+                        break
```

Comment:
According to the [docs](https://docs.python-requests.org/en/latest/user/advanced/#body-content-workflow):
`If you set stream to True when making a request, Requests cannot release the connection back to the pool unless you consume all the data or call [Response.close](https://docs.python-requests.org/en/latest/api/#requests.Response.close).`

Perhaps we should close the connection in this case

Reply:
This should not be an issue, given that the request is made using a `with` statement (see line 127 of fetcher.py) as recommended by the docs: `If you find yourself partially reading request bodies (or not reading them at all) while using stream=True, you should make the request within a with statement to ensure it’s always closed`

Reply:
Ah, perfect, thanks for the explanation, somehow missed the fact that you used `with` statement